src/parser.ts

```diff
@@ -413,19 +412,15 @@ function getLeftHandSideName(left: ts.Expression): string {
 }
 
 function getPropertyName(name: ts.PropertyName): string {
-    if (
-        ts.isIdentifier(name) ||
-        ts.isStringLiteral(name) ||
-        ts.isNumericLiteral(name)
-    ) {
-        return name.text;
+    switch (name.kind) {
+        case ts.SyntaxKind.Identifier:
+        case ts.SyntaxKind.StringLiteral:
+        case ts.SyntaxKind.NumericLiteral:
+        case ts.SyntaxKind.PrivateIdentifier:
+            return name.text;
+        case ts.SyntaxKind.ComputedPropertyName:
+            return computedName(name.expression);
     }
-    if (ts.isComputedPropertyName(name)) {
-        return computedName(name.expression);
-    }
-    // all cases are covered above, should never reach here
-    assert(false);
```

Comment:
This is justified by leaning into TypeScript exhaustiveness checking in the `switch`. Removing one of the `case` statements yields a TypeScript error.